python/lsst/sims/catUtils/utils/LightCurveGenerator.py

```diff
@@ -0,0 +1,496 @@
+from __future__ import print_function
+import numpy as np
+import copy
+
+from lsst.sims.catUtils.utils import ObservationMetaDataGenerator
+from lsst.sims.catUtils.mixins import PhotometryStars, VariabilityStars
+from lsst.sims.catUtils.mixins import PhotometryGalaxies, VariabilityGalaxies
+from lsst.sims.catalogs.measures.instance import InstanceCatalog, compound, cached
+from lsst.sims.utils import haversine
+
+import time
+
+__all__ = ["StellarLightCurveGenerator", "AgnLightCurveGenerator",
+           "_baseLightCurveCatalog", "LightCurveGenerator"]
+
+# a global cache to store SedLists loaded by the light curve catalogs
+_sed_cache = {}
+
+
+class _baseLightCurveCatalog(InstanceCatalog):
+    """
+    """
+
+    column_outputs = ["uniqueId", "raJ2000", "decJ2000",
+                      "lightCurveMag", "sigma_lightCurveMag",
+                      "truthInfo"]
+
+    def iter_catalog(self, chunk_size=None, query_cache=None):
+        """
+        Returns an iterator over rows of the catalog.
+
+        Parameters
+        ----------
+        chunk_size : int, optional, defaults to None
+            the number of rows to return from the database at a time. If None,
+            returns the entire database query in one chunk.
+
+        query_cache : iterator over database rows, optional, defaults to None
+            the result of calling db_obj.query_columns().  If query_cache is not
+            None, this method will iterate over the rows in query_cache and produce
+            an appropriate InstanceCatalog. DO NOT use this
+            unless you know what you are doing.  It is an optional
```

Comment:
I agree: only advanced users should try to write their own LightCurveGenerators.  I guess I wanted the API to look as much like the "normal" iter_catalog() as possible so that those advanced users (who are presumably familiar with InstanceCatalog) can intuit what is going on.

I will change the docstring as you suggest.